pkg/operator/status.go

```diff
@@ -525,83 +501,33 @@ func (optr *Operator) cfeEvalCgroupsV1() (bool, error) {
-		return skewUnchecked, coStatus, err
+// checkNodeUpgradeable checks current Node status to look for anything incompatible with the next 4.(y+1) OpenShift release.
+func (optr *Operator) checkNodeUpgradeable(pools []*mcfgv1.MachineConfigPool) (coStatus configv1.ClusterOperatorStatusCondition, err error) {
+	coStatus = configv1.ClusterOperatorStatusCondition{
+		Status: configv1.ConditionTrue,
 	}
-	var (
-		lastError      error
-		kubeletVersion string
-	)
 	nodes, err := optr.GetAllManagedNodes(pools)
 	if err != nil {
 		err = fmt.Errorf("getting all managed nodes failed: %w", err)
-		coStatus.Reason = skewUnchecked
-		coStatus.Message = fmt.Sprintf("An error occurred when getting all the managed nodes: %v", err.Error())
+		coStatus.Status = convigv1.ConditionUnknown
+		coStatus.Reason = "FailedToGetNodes"
+		coStatus.Message = err.Error()
+		return coStatus, err
 	}
+	rhelNodes := make([]string, 0, len(nodes))
 	for _, node := range nodes {
-		// looks like kubeletVersion: v1.21.0-rc.0+6143dea
-		kubeletVersion = node.Status.NodeInfo.KubeletVersion
-		if kubeletVersion == "" {
-			continue
-		}
-		nodeMinorVersion, err := getMinorKubeletVersion(kubeletVersion)
-		if err != nil {
-			lastError = err
-			continue
+		osImage := node.Status.NodeInfo.OSImage
+		if strings.HasPrefix(osImage, "Red Hat Enterprise Linux") && !strings.HasPrefix(osImage, "Red Hat Enterprise Linux CoreOS") {
+			rhelNodes = append(rhelNodes, fmt.Sprintf("%s (%s)", node.Name, osImage))
 		}
-		if nodeMinorVersion+2 < kubeAPIServerMinorVersion {
-			coStatus.Reason = skewUnsupported
-			coStatus.Message = fmt.Sprintf("One or more nodes have an unsupported kubelet version skew. Please see `oc get nodes` for details and upgrade all nodes so that they have a kubelet version of at least %v.", getMinimalSkewSupportNodeVersion(kubeAPIServerVersion))
-			return skewUnsupported, coStatus, nil
-		}
-		if nodeMinorVersion+2 == kubeAPIServerMinorVersion {
-			coStatus.Reason = skewPresent
-			coStatus.Message = fmt.Sprintf("Current kubelet version %v will not be supported by newer kube-apiserver. Please upgrade the kubelet first if plan to upgrade the kube-apiserver", kubeletVersion)
-			return skewPresent, coStatus, nil
-		}
-	}
-	if kubeletVersion == "" {
-		err = fmt.Errorf("kubelet does not yet have a version")
-		coStatus.Reason = skewUnchecked
-		coStatus.Message = fmt.Sprintf("An error occurred when checking kubelet version skew: %v", err.Error())
-		return skewUnchecked, coStatus, err
 	}
-	if lastError != nil {
-		coStatus.Reason = skewUnchecked
-		coStatus.Message = fmt.Sprintf("An error occurred when checking kubelet version skew: %v", err)
-		return skewUnchecked, coStatus, lastError
+	strings.Sort(rhelNodes)
+	if len(rhelNodes) > 0 {
+		coStatus.Status = convigv1.ConditionFalse
+		coStatus.Reason = "RHELNodes"
+		coStatus.Message = fmt.Sprintf("%s RHEL nodes, including %s, but OpenShift 4.19 requires RHCOS https://FIXME-DOC-LINK", len(rhelNodes), rhelNodes[0])
```

Comment:
Any ideas where I should be linking folks running 4.18 today for "bare-RHEL is gone in 4.19, and here's how we recommend you migrate..."?

Reply:
OSDOCS-12979 is probably for this. cc @gpei

Reply:
@jhou1 yes, that's the Doc jira issue tracking it

Reply:
https://docs.redhat.com/en/documentation/openshift_container_platform/4.18/html/updating_clusters/preparing-to-update-a-cluster#updating-cluster-prepare-past-4-18 is live via openshift/openshift-docs#92638 and openshift/openshift-docs#92737.  I've updated the link in this line with 56ba6cb08 -> 13cceb069, and I've dropped the `WIP` from this pull's title, and I'm marking this thread resolved.